### PR TITLE
fix(web): Broaden URL match to catch organization slug

### PR DESCRIPTION
**Pull request by RyanSkonnord**
For logged-out users in an organization that requires SSO, it is
necessary to match the "organization_slug" capturing group in order to
correctly redirect to the organization-specific login page.

Broaden the catch-all mapping to react_page_view so that paths not
covered elsewhere, such as `/organizations/{org}/discover/`, redirect to
login correctly.

### CODE REVIEW

**Comment by RyanSkonnord** [comment]
For https://getsentry.atlassian.net/browse/ER-587. Continues from #25200.

File: src/sentry/web/urls.py [hunk 1]
```diff
@@ -536,8 +536,8 @@
                     RestoreOrganizationView.as_view(),
                     name="sentry-restore-organization",
                 ),
-                # need to catch settings and force it to react
-                url(r"^(?P<organization_slug>[\w_-]+)/settings/", react_page_view),
+                # need to catch `settings`, `discover`, etc., and force them to react
+                url(r"^(?P<organization_slug>[\w_-]+)/[\w_-]+/", react_page_view),
```
Comment (EvanPurkhiser) [inline src/sentry/web/urls.py:540]: This feels pretty broad. Is it too broad?

Comment (markstory) [reply src/sentry/web/urls.py:540]: I think it being broad is what helps get SSO users to the right place when they don't have an active session though.

Comment (RyanSkonnord) [reply src/sentry/web/urls.py:540]: Yes.

If this regex catches some nonsense address that it wouldn't have previously, logged-in users will still see the 404 page from `react_page_view`. Logged-out users would be taken to a login page either way (and would be redirected back to the 404 view after validating); this only affects whether they see the org SSO page.

Making it more specific (such as `r"^(?P<organization_slug>[\w_-]+)/(settings|discover)/"`) would, AFIACT, only have the effect of causing SSO users to see their org's login page if they land on an address that exists and the non-SSO login page if they land on one that doesn't.

Comment (markstory) [reply src/sentry/web/urls.py:540]: Having a more specific pattern is going to be a long term maintenance headache as well.

**Review by markstory — APPROVED** [review]

**Review by leedongwei — APPROVED** [review]

**Comment by RyanSkonnord** [comment]
See https://github.com/getsentry/sentry/pull/25470/files#diff-94c34ada075eea4de39d2e02b21b1826f4c3f1753cc3b3761429f2b5cfd7b28aR81-R86 regarding my resolution to the CI failure.

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**